client/question/domain/PrereqCheckErrorObjectFactory.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating a PrereqCheckError of a student's TIE
+ * session.
+ */
+
+tie.factory('PrereqCheckErrorObjectFactory', [
+  function() {
+    /**
+     * PrereqCheckError objects are used to store information related to Prereq
+     errors.
+     */
+
+    /**
+     * Constructor for PrereqCheckError.
+     *
+     * @constructor
+     */
+    var PrereqCheckError = function(errorName, errorLineNumber) {
+      /**
+       * String standing for key in WRONG_LANGUAGE_ERRORS that correlates to
+       * error that triggered.
+       * Should be null if error type is not 'wrong language.'
+       *
+       * @type {string | null}
+       * @private
+       */
+      this._errorName = errorName;
+
+      /**
+       * Line number that correlates to error that triggered.
+       * Should be null if error type does not supply a line number.
+       *
+       * @type {integer | null}
+       * @private
+       */
+      this._errorLineNumber = errorLineNumber;
+    };
+
+    /**
+     * A getter for the _errorName property.
+     *
+     * @returns {string}
+     */
+    PrereqCheckError.prototype.getErrorName = function() {
+      return this._errorName;
+    };
+
+    /**
+     * A setter for the _errorName property.
+     *
+     * @param {string} newName The key to set the _errorName property to.
+     */
+    PrereqCheckError.prototype.setErrorName = function(newName) {
+      this._errorName = newName;
+    };
+
+    /**
+     * A getter for the _errorLineNumber property.
+     *
+     * @returns {string}
+     */
+    PrereqCheckError.prototype.getErrorLineNumber = function() {
+      return (this._errorLineNumber < 0) ? null : this._errorLineNumber;
+    };
+
+    /**
+     * A setter for the _errorLineNumber property.
+     *
+     * @param {integer} errorLine to set the _errorLineNumber property to.
+     */
+    PrereqCheckError.prototype.setErrorLineNumber = function(newLineNumber) {
+      this._errorLineNumber = newLineNumber;
+    };
+
+    // Static class methods.
+    /**
+     * Returns a new PrereqCheckError object.
+     *
+     * @returns {PrereqCheckError}
+     */
+    PrereqCheckError.create = function(errorName, errorLineNumber) {
+      if (errorLineNumber < 0 && errorName === 'notOp') {
```

Comment:
I don't think I understand this, it might need an explanatory comment. What's special about 'notOp'?